### PR TITLE
Ran `black`

### PR DESCRIPTION
**Pull request by teran-mckinney**
black, version 19.10b0

No other changes.

#102

### CODE REVIEW

**Comment by teran-mckinney** [comment]
@ofek, could you take a look?

**Comment by ofek** [comment]
Oh dude I suck, sorry! Will today

**Comment by teran-mckinney** [comment]
Ha, it's okay! Thank you!

**Comment by teran-mckinney** [comment]
Is this better?

File: bit/base58.py [hunk 1]
```diff
@@ -49,8 +49,9 @@ def b58decode(string):
             num *= 58
             num += alphabet_index[char]
     except KeyError:
-        raise ValueError('"{}" is an invalid base58 encoded '
-                         'character.'.format(char)) from None
+        raise ValueError(
+            '"{}" is an invalid base58 encoded ' 'character.'.format(char)
```
Comment (ofek) [inline bit/base58.py:53]:
```suggestion
            '"{}" is an invalid base58 encoded character.'.format(char)
```

File: bit/keygen.py [hunk 2]
```diff
@@ -19,8 +19,9 @@ def generate_matching_address(prefix, cores='all'):  # pragma: no cover
 
     for char in prefix:
         if char not in BASE58_ALPHABET:
-            raise ValueError('{} is an invalid base58 encoded '
-                             'character.'.format(char))
+            raise ValueError(
+                '{} is an invalid base58 encoded ' 'character.'.format(char)
```
Comment (ofek) [inline bit/keygen.py:23]:
```suggestion
                '{} is an invalid base58 encoded character.'.format(char)
```

File: bit/keygen.py [hunk 3]
```diff
@@ -66,9 +66,7 @@ def generate_matching_address(prefix, cores='all'):  # pragma: no cover
         sys.stdout.flush()
 
     private_key, address = queue.get()
-    print('\n\n'
-          'WIF: {}\n'
-          'Address: {}'.format(bytes_to_wif(private_key), address))
+    print('\n\n' 'WIF: {}\n' 'Address: {}'.format(bytes_to_wif(private_key), address))
```
Comment (ofek) [inline bit/keygen.py:69]:
```suggestion
    print('\n\nWIF: {}\nAddress: {}'.format(bytes_to_wif(private_key), address))
```

File: bit/network/services.py [hunk 4]
```diff
@@ -652,8 +728,9 @@ def broadcast_tx(cls, tx_hex):  # pragma: no cover
                 pass
 
         if success is False:
-            raise ConnectionError('Transaction broadcast failed, or '
-                                  'Unspents were already used.')
+            raise ConnectionError(
+                'Transaction broadcast failed, or ' 'Unspents were already used.'
```
Comment (ofek) [inline bit/network/services.py:732]:
```suggestion
                'Transaction broadcast failed, or Unspents were already used.'
```

File: bit/network/services.py [hunk 5]
```diff
@@ -677,7 +754,8 @@ def broadcast_tx_testnet(cls, tx_hex):  # pragma: no cover
...
```
Comment (ofek) [inline bit/network/services.py:758]:
```suggestion
                'Transaction broadcast failed, or Unspents were already used.'
```

File: bit/wallet.py [hunk 6]
```diff
@@ -896,11 +975,14 @@ class MultiSig:
     def __init__(self, private_key, public_keys, m):
 
         if private_key.instance != 'PrivateKey':
-            raise TypeError('MultiSig only accepts a PrivateKey class to '
-                            'assign a private key.')
-
-        if (bytes_to_hex(private_key.public_key) not in public_keys
-                and private_key.public_key not in public_keys):
+            raise TypeError(
+                'MultiSig only accepts a PrivateKey class to ' 'assign a private key.'
```
Comment (ofek) [inline bit/wallet.py:979]:
```suggestion
                'MultiSig only accepts a PrivateKey class to assign a private key.'
```

File: bit/base58.py (hunk 1 above)
Comment (teran-mckinney) [reply bit/base58.py:53]: Thanks for pointing that out. Really weird.

**Comment by teran-mckinney** [comment]
I submitted this: https://github.com/psf/black/issues/1338

Not sure if this is okay to clean up one time. `black`'s master branch has the same issue.

Will wait at least a couple days to hear back from the developers and see.

I guess this is not as straight forward as I thought. I was hoping it would be more bulletproof like `go fmt`.

**Comment by ofek** [comment]
Yup just fix these

**Comment by teran-mckinney** [comment]
So I ran `black` again and it changed a bunch more. Maybe I should have done two passes the first time around.

After the latest one, `black` output seems stable at least.

**Comment by ofek** [comment]
https://github.com/ofek/bit/pull/103#pullrequestreview-389592811

**Comment by ofek** [comment]
Revert, then maybe add https://github.com/ofek/coincurve/blob/master/pyproject.toml (with py36 = true)

File: bit/keygen.py [hunk 7]
```diff
@@ -19,8 +19,7 @@ def generate_matching_address(prefix, cores='all'):  # pragma: no cover
 
     for char in prefix:
         if char not in BASE58_ALPHABET:
-            raise ValueError('{} is an invalid base58 encoded '
-                             'character.'.format(char))
+            raise ValueError('{} is an invalid base58 encoded ' 'character.'.format(char))
```
Comment (ofek) [inline bit/keygen.py:22]:
```suggestion
            raise ValueError('{} is an invalid base58 encoded character.'.format(char))
```

File: bit/network/services.py [hunk 8]
```diff
@@ -652,8 +642,7 @@ def broadcast_tx(cls, tx_hex):  # pragma: no cover
                 pass
 
         if success is False:
-            raise ConnectionError('Transaction broadcast failed, or '
-                                  'Unspents were already used.')
+            raise ConnectionError('Transaction broadcast failed, or ' 'Unspents were already used.')
```
Comment (ofek) [inline bit/network/services.py:645]:
```suggestion
            raise ConnectionError('Transaction broadcast failed, or Unspents were already used.')
```

File: bit/network/services.py [hunk 9]
```diff
@@ -677,7 +666,6 @@ def broadcast_tx_testnet(cls, tx_hex):  # pragma: no cover
...
```
Comment (ofek) [inline bit/network/services.py:669]:
```suggestion
            raise ConnectionError('Transaction broadcast failed, or Unspents were already used.')
```

File: bit/transaction.py [hunk 10]
```diff
@@ -465,8 +438,7 @@ def sanitize_tx_data(unspents, outputs, fee, leftover, combine=True,
         dest, amount = output
         vs = get_version(dest)
         if vs and vs != version:
-            raise ValueError('Cannot send to ' + vs + 'net address when '
-                             'spending from a ' + version + 'net address.')
+            raise ValueError('Cannot send to ' + vs + 'net address when ' 'spending from a ' + version + 'net address.')
```
Comment (ofek) [inline bit/transaction.py:441]:
```suggestion
            raise ValueError('Cannot send to ' + vs + 'net address when spending from a ' + version + 'net address.')
```

File: bit/transaction.py [hunk 11]
```diff
@@ -607,15 +573,15 @@ def sign_tx(private_key, tx, *, unspents):
...
                 continue
-            tx_input = hex_to_bytes(unspent.txid)[::-1] + \
-                unspent.txindex.to_bytes(4, byteorder='little')
+            tx_input = hex_to_bytes(unspent.txid)[::-1] + unspent.txindex.to_bytes(4, byteorder='little')
             input_dict[tx_input] = unspent.to_dict()
     except TypeError:
-        raise TypeError('Please provide as unspents at least all inputs to '
-                        'be signed with the function call in a list.')
+        raise TypeError(
+            'Please provide as unspents at least all inputs to ' 'be signed with the function call in a list.'
```
Comment (ofek) [inline bit/transaction.py:580]:
```suggestion
            'Please provide as unspents at least all inputs to be signed with the function call in a list.'
```

File: bit/transaction.py [hunk 12]
```diff
@@ -692,8 +657,7 @@ def sign_tx(private_key, tx, *, unspents):
                             # we just overwrite it and don't care.
                             sigs[pub] = sig
                 if len(sigs) >= private_key.m:
-                    raise ValueError('Transaction is already signed with '
-                                     'sufficiently needed signatures.')
+                    raise ValueError('Transaction is already signed with ' 'sufficiently needed signatures.')
```
Comment (ofek) [inline bit/transaction.py:660]:
```suggestion
                    raise ValueError('Transaction is already signed with sufficiently needed signatures.')
```

File: bit/wallet.py [hunk 13]
```diff
@@ -290,12 +297,10 @@ def create_transaction(self, outputs, fee=None, absolute_fee=False,
         try:
             unspents = unspents or self.get_unspents()
         except ConnectionError:
-            raise ConnectionError('All APIs are unreachable. Please provide '
-                                  'the unspents to spend from directly.')
+            raise ConnectionError('All APIs are unreachable. Please provide ' 'the unspents to spend from directly.')
```
Comment (ofek) [inline bit/wallet.py:300]:
```suggestion
            raise ConnectionError('All APIs are unreachable. Please provide the unspents to spend from directly.')
```

File: bit/wallet.py [hunk 14]
```diff
@@ -659,12 +681,10 @@ def create_transaction(self, outputs, fee=None, absolute_fee=False,
...
```
Comment (ofek) [inline bit/wallet.py:684]:
```suggestion
            raise ConnectionError('All APIs are unreachable. Please provide the unspents to spend from directly.')
```

File: bit/wallet.py [hunk 15]
```diff
@@ -896,11 +933,9 @@ class MultiSig:
     def __init__(self, private_key, public_keys, m):
 
         if private_key.instance != 'PrivateKey':
-            raise TypeError('MultiSig only accepts a PrivateKey class to '
-                            'assign a private key.')
+            raise TypeError('MultiSig only accepts a PrivateKey class to ' 'assign a private key.')
```
Comment (ofek) [inline bit/wallet.py:936]:
```suggestion
            raise TypeError('MultiSig only accepts a PrivateKey class to assign a private key.')
```

File: bit/wallet.py [hunk 16]
```diff
@@ -1064,12 +1101,10 @@ def create_transaction(self, outputs, fee=None, absolute_fee=False,
         try:
             unspents = unspents or self.get_unspents()
         except ConnectionError:
-            raise ConnectionError('All APIs are unreachable. Please provide '
-                                  'the unspents to spend from directly.')
+            raise ConnectionError('All APIs are unreachable. Please provide ' 'the unspents to spend from directly.')
```
Comment (ofek) [inline bit/wallet.py:1104]:
```suggestion
            raise ConnectionError('All APIs are unreachable. Please provide the unspents to spend from directly.')
```

File: bit/wallet.py [hunk 17]
```diff
@@ -1202,11 +1245,9 @@ class MultiSigTestnet:
     def __init__(self, private_key, public_keys, m):
 
         if private_key.instance != 'PrivateKeyTestnet':
-            raise TypeError('MultiSigTesnet only accepts PrivateKeyTestnet '
-                            'class to assign a private key.')
+            raise TypeError('MultiSigTesnet only accepts PrivateKeyTestnet ' 'class to assign a private key.')
```
Comment (ofek) [inline bit/wallet.py:1248]:
```suggestion
            raise TypeError('MultiSigTesnet only accepts PrivateKeyTestnet class to assign a private key.')
```

File: bit/wallet.py [hunk 18]
```diff
@@ -1369,12 +1416,10 @@ def create_transaction(self, outputs, fee=None, absolute_fee=False,
         try:
             unspents = unspents or self.get_unspents()
         except ConnectionError:
-            raise ConnectionError('All APIs are unreachable. Please provide '
-                                  'the unspents to spend from directly.')
+            raise ConnectionError('All APIs are unreachable. Please provide ' 'the unspents to spend from directly.')
```
Comment (ofek) [inline bit/wallet.py:1419]:
```suggestion
            raise ConnectionError('All APIs are unreachable. Please provide the unspents to spend from directly.')
```

**Comment by teran-mckinney** [comment]
Thanks for catching those. I should have just used `grep " ' '"` to catch the rest.

**Comment by teran-mckinney** [comment]
Awesome, thank you!

**Pull request merged**